source/fundamentals/monitoring.txt

```diff
@@ -1,7 +1,435 @@
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Value
+     - Description
+   * - ``0``
+     - Unknown instance
+   * - ``1``
+     - Standalone instance
+   * - ``2``
+     - Replica set member
+   * - ``6``
+     - Primary instance
+   * - ``10``
+     - Secondary instance
+   * - ``18``
+     - Arbiter instance
+   * - ``34``
+     - Replica set ghost, a member that cannot be queried
+   * - ``256``
+     - Mongos proxy instance
+   * - ``512``
+     - Load balancer instance
+
+ServerOpeningEvent
+~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+   :copyable: false
+
+   *event.ServerOpeningEvent
+   {
+       "Address": "...",
+       "TopologyID": "..."
+   }
+
+ServerClosedEvent
+~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+   :copyable: false
+
+   *event.ServerClosedEvent
+   {
+       "Address": "...",
+       "TopologyID": "..."
+   }
+
+TopologyDescriptionChangedEvent
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. important::
+
+   The driver calls ``TopologyDescriptionChanged`` when the deployment topology is locked,
+   so the callback should not attempt any operation that requires server
+   selection on the same client.
+
+.. code-block:: bash
+   :copyable: false
+
+   *event.TopologyDescriptionChangedEvent
+   {
+       "TopologyID": "...",
+       "PreviousDescription": {
+           "Servers": [
+               {
+                   "Addr": "...",
+                   "Arbiters": null,
+                   "AverageRTT": 0,
+                   "AverageRTTSet": false,
+                   "Compression": null,
+                   "CanonicalAddr": "...",
+                   "ElectionID": "...",
+                   "HeartbeatInterval": 0,
+                   "HelloOK": false,
+                   "Hosts": null,
+                   "LastError": null,
+                   "LastUpdateTime": "...",
+                   "LastWriteTime": "...",
+                   "MaxBatchCount": 0,
+                   "MaxDocumentSize": 0,
+                   "MaxMessageSize": 0,
+                   "Members": null,
+                   "Passives": null,
+                   "Passive": false,
+                   "Primary": "...",
+                   "ReadOnly": false,
+                   "ServiceID": null,
+                   "SessionTimeoutMinutes": 0,
+                   "SetName": "...",
+                   "SetVersion": 0,
+                   "Tags": null,
+                   "TopologyVersion": null,
+                   "Kind": 0,
+                   "WireVersion": null
+               },
+               ...
+           ],
+           "SetName": "...",
+           "Kind": 10,
+           "SessionTimeoutMinutes": 30,
+           "CompatibilityErr": null
+       },
+       "NewDescription": {
+           "Servers": [...],
+           "SetName": "...",
+           "Kind": 10,
+           "SessionTimeoutMinutes": 30,
+           "CompatibilityErr": null
+       }
+   }
+
+The ``Kind`` field in the server descriptions represents the type of a
```

Comment:
Q: I see that this table is repeated throughout the page. Is there a reasonably clean way to include it just once? Maybe a subheading dedicated to this field?

Reply:
I used the Node page as a reference and it is repeated there, I have converted it to a note and added a reference.

Reply:
actually a note made it seem like it was applicable to the section it was under. made it a subheading.